app/Core/Ldap/User.php

```diff
@@ -88,16 +88,34 @@ public function find($query)
      * @param  string  $username
      * @return string[]
      */
+    protected function getGroupsRecursive(Entry $entry, $member, &$groupsAll)
+    {
+
+        $groups = $this->group->find(sprintf($this->getGroupUserFilter(), $member));
+
+        foreach ($groups as $group) {
+            $groupId = $group->getExternalId();
+
+            if (!isset($groupsAll[$groupId])) {
+                $groupsAll[$groupId] = $member;
+                $this->getGroupsRecursive($entry, $groupId, $groupsAll);
```

Comment:
The use of the isset( groupsAll ) + reference passing is there to avoid processing a group that has already been processed (or is in progress, as registered before the recursion call) thus avoiding infinite recursion.

That would be nice to do a test with it nonetheless, but I don't know phpunit and the current tests well enough to do it atm.